### ISSUE TITLE
SingleLineMethods + AllowIfMethodIsEmpty:false raises during autocorrect

### ISSUE THREAD

**Issue by matthewd** [bbatsov/rubocop#2286]
```
0.34.2 (using Parser 2.2.2.6, running on ruby 2.2.2 x86_64-darwin14)
```

Autocorrecting an empty single-line method:
```ruby
def x; end
```

Raises:
```
% rubocop -c <(<<<"SingleLineMethods: { AllowIfMethodIsEmpty: false }") -a <(<<<"def x; end")
Inspecting 1 file


0 files inspected, no offenses detected
undefined method `type' for nil:NilClass
lib/rubocop/cop/style/single_line_methods.rb:37:in `block in autocorrect'
lib/rubocop/cop/corrector.rb:46:in `call'
lib/rubocop/cop/corrector.rb:46:in `block in rewrite'
lib/rubocop/cop/corrector.rb:45:in `each'
lib/rubocop/cop/corrector.rb:45:in `rewrite'
lib/rubocop/cop/team.rb:90:in `autocorrect_one_cop'
lib/rubocop/cop/team.rb:65:in `autocorrect'
lib/rubocop/cop/team.rb:35:in `inspect_file'
lib/rubocop/runner.rb:170:in `inspect_file'
lib/rubocop/runner.rb:141:in `block in do_inspection_loop'
lib/rubocop/runner.rb:135:in `loop'
lib/rubocop/runner.rb:135:in `do_inspection_loop'
lib/rubocop/runner.rb:87:in `process_file'
lib/rubocop/runner.rb:57:in `block in inspect_files'
lib/rubocop/runner.rb:55:in `each'
lib/rubocop/runner.rb:55:in `inspect_files'
lib/rubocop/runner.rb:33:in `run'
lib/rubocop/cli.rb:26:in `run'
```
